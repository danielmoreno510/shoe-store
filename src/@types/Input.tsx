type InputProps = {
  style: any;
  value: string;
  placeholder: string;
  onChangeText(value: 'string' | any): any;
  onFocus(): any;
  onBlur(): any;
  keyboardType?: any;
  secureTextEntry: boolean;
};
