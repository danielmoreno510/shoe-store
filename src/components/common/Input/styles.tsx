import {StyleSheet} from 'react-native';

import {primaryColor} from '../../../constants';

export default StyleSheet.create({
  input: {
    borderBottomColor: primaryColor,
    borderBottomWidth: 2,
    fontSize: 18,
  },
});
