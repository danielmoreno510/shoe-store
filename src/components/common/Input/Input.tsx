import React from 'react';
import {TextInput} from 'react-native';

import styles from './styles';

const CustomInput: React.FC<InputProps> = ({
  style,
  value,
  placeholder,
  onChangeText,
  onFocus,
  onBlur,
  keyboardType,
  secureTextEntry,
}) => {
  return (
    <TextInput
      style={[style, styles.input]}
      placeholder={placeholder}
      value={value}
      onChangeText={onChangeText}
      onFocus={onFocus}
      onBlur={onBlur}
      secureTextEntry={secureTextEntry}
      keyboardType={keyboardType}
      placeholderTextColor={'#aaaaaa'}
    />
  );
};

export default CustomInput;
