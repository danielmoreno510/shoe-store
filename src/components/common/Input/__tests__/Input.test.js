import React from 'react';
import {shallow} from 'enzyme';

import Input from '../Input';

describe('Input component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      style: {},
      value: 'test',
      placeholder: 'test',
      onChangeText: () => {},
      onFocus: () => {},
      onBlur: () => {},
      keyboardType: 'default',
      secureTextEntry: true,
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<Input {...testProps} />);

      it('THEN should display a regular Input', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });
  });
});
