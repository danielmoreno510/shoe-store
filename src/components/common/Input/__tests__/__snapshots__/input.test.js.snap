// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Input 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="default"
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  placeholder="test"
  placeholderTextColor="#aaaaaa"
  rejectResponderTermination={true}
  secureTextEntry={true}
  style={
    Array [
      Object {},
      Object {
        "borderBottomColor": "#f1b30a",
        "borderBottomWidth": 2,
        "fontSize": 18,
      },
    ]
  }
  underlineColorAndroid="transparent"
  value="test"
/>
`;
